tool/gravity/cli/system.go

```diff
@@ -362,57 +356,105 @@ func applyUpdates(env *localenv.LocalEnvironment, updates []storage.PackageUpdat
-	secrets, err := findSecretsPackage(packages)
+	secretsPackage, err := pack.FindSecretsPackage(packages)
 	if err != nil {
-		return nil, trace.Wrap(err, "failed to find secrets package")
+		return nil, trace.Wrap(err, "failed to find installed secrets package")
 	}
 
-	err = updateInstalledLabelIfNecessary(packages, secrets.Locator)
+	installedSecretsPackage, err := pack.FindInstalledPackage(packages, *secretsPackage)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, trace.Wrap(err, "failed to find installed secrets package")
 	}
 
-	planetPackage, planetConfig, err := findRuntimePackageWithConfig(packages)
+	existingRuntime, existingRuntimeConfig, err := pack.FindRuntimePackageWithConfig(packages)
 	if err != nil {
 		return nil, trace.Wrap(err, "failed to find runtime package")
 	}
-	log.Debugf("Found existing runtime package %v with configuration package %v.",
-		*planetPackage, *planetConfig)
+	log.WithFields(logrus.Fields{
+		"runtime": existingRuntime,
+		"config":  existingRuntimeConfig,
+	}).Info("Found existing runtime and configuration packages.")
+
+	runtimeConfigUpdate, err := maybeConvertLegacyPlanetConfigPackage(*existingRuntimeConfig)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 
-	planetConfigUpdate, err := maybeConvertLegacyPlanetConfigPackage(*planetConfig)
+	updateGravityPackage, err := newPackageRequest(packages, gravityPackageFilter)
 	if err != nil {
-		return nil, trace.Wrap(err, "failed to find runtime configuration package")
+		return nil, trace.Wrap(err, "failed to find installed gravity binary package")
+	}
+
+	existingTeleport, existingTeleportConfig, err := pack.FindTeleportPackageWithConfig(packages)
+	if err != nil {
+		return nil, trace.Wrap(err)
 	}
+	log.WithFields(logrus.Fields{
+		logrus.ErrorKey: err,
+		"package":       existingTeleport,
+		"config":        existingTeleportConfig,
+	}).Info("Found existing teleport and configuration packages.")
 
 	reqs = append(reqs,
-		newPackageRequest(secrets.Locator),
+		*updateGravityPackage,
 		packageRequest{
-			filter:       *planetPackage,
-			updateFilter: &runtimePackageUpdate,
-			labels:       pack.RuntimePackageLabels,
+			installedPackage: *installedSecretsPackage,
+			labels:           pack.RuntimeSecretsPackageLabels,
 		},
 		packageRequest{
-			filter: *planetConfig,
-			// Look for updated package name in upstream packages
-			updateFilter:          planetConfigUpdate,
-			withoutInstalledLabel: true,
+			installedPackage: *existingRuntime,
+			updatePackage:    &runtimePackageUpdate,
+			labels:           pack.RuntimePackageLabels,
+			configPackage: &packageRequest{
+				installedPackage: *existingRuntimeConfig,
+				updateFilter:     runtimeConfigUpdate,
+				labels: pack.ConfigLabels(
+					*existingRuntimeConfig,
+					pack.PurposePlanetConfig,
+				),
+				less: configPackageLess,
+			},
+		},
+		packageRequest{
+			installedPackage: *existingTeleport,
```

Comment:
IIRC teleport node config package was being pulled in one of the update phases. Did it get moved to system update command?

Reply:
No, it did not - it is duplicating this logic and should be idempotent.